kube-runtime/src/wait.rs

```diff
@@ -192,6 +193,75 @@ pub mod conditions {
         }
     }
 
+    /// An await condition that returns `true` if the object exists.
+    ///
+    /// NOTE: If waiting for an object to be deleted, do _not_ [invert](`Condition::not`) this [`Condition`].
+    /// Instead, use [`is_deleted`], which considers a deleted-then-recreated object to have been deleted.
+    #[must_use]
+    pub fn exists<K>() -> impl Condition<K> {
+        |obj: Option<&K>| obj.is_some()
+    }
+
+    /// An await condition for [`Resource`] that returns `true` if the object's condition of the given type holds true
+    /// for the given `value_cond`.
+    ///
+    /// # Value condition
+    ///
+    /// The value condition is passed `None` if the object does not exist or does not have the given condition (combine with
+    /// [`exists`] if you need to validate whether the object exists). Otherwise, the value should be one of `"True"`, `"False"`, or `"Unknown"`
+    /// (see <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#condition-v1-meta> for more details).
+    ///
+    /// # Stability
+    ///
+    /// This is an experimental API that should be expected to change. It has a few particular problems:
+    ///
+    /// 1. It is completely untyped
+    /// 2. It makes fairly deep assumptions about the structure of the object and its status
+    /// 3. It doesn't have any way to signal errors gracefully
+    /// 4. It has some unfortunate lifetime problems that prevent bringing in a closure context
+    ///
+    /// # Usage
+    ///
+    /// ```rust
+    /// # use k8s_openapi::api::core::v1::{Pod, PodCondition, PodStatus};
+    /// # use kube_runtime::wait::{conditions::unstable_has_status_condition, Condition};
+    /// let pod = |ready: String| Pod {
+    ///     status: Some(PodStatus {
+    ///         conditions: Some(vec![PodCondition {
+    ///             type_: "Ready".to_string(),
+    ///             status: ready,
+    ///             ..PodCondition::default()
+    ///         }]),
+    ///         ..PodStatus::default()
+    ///     }),
+    ///     ..Pod::default()
+    /// };
+    /// let cond_status_ready: fn(Option<&str>) -> bool = |status| status == Some("True");
+    /// let cond_pod_ready = unstable_has_status_condition("Ready", cond_status_ready);
+    /// assert!(!cond_pod_ready.matches_object(Some(&pod("False".to_string()))));
+    /// assert!(cond_pod_ready.matches_object(Some(&pod("True".to_string()))));
+    /// ```
+    #[must_use]
+    pub fn unstable_has_status_condition<'a, K: Serialize + Resource, StatusCond: Condition<str> + 'a>(
```

Comment:
That kind of makes sense. I'd almost rather have this be something that the end user has to opt into specifically with a `--cfg`, rather than a feature you can inherit by accident from a dependency.

Reply:
There is the tokio solution for this as well https://github.com/kube-rs/kube-rs/issues/508#issuecomment-946017805. Anyway, I've made notes about this in the big stability roadmap issue, because it's something we should solve for that.